File: 백준/Silver/15650. N과 M （2）/N과 M （2）.js
const [N, M] = require('fs').readFileSync(0, 'utf-8').toString().trim().split(' ').map(Number);
const dx = Array.from(Array(N), (_, i) => i + 1);
let answer = '';

function dfs(x, sequence) {
    if (sequence.length === M) {
        answer += `${sequence.join(' ')}\n`;
        return;
    }
    
    const maxNum = sequence[sequence.length - 1];
    
    for (let nx of dx) {
        if (nx <= maxNum) continue;
        
        sequence.push(nx);
        dfs(nx, sequence);
        sequence.pop();
    }
}

for (let x of dx) {
    const sequence = [x];
    
    dfs(x, sequence);
}

console.log(answer);

File: 백준/Silver/15650. N과 M （2）/N과 M （2）.py
N, M = map(int, input().split())
visited = set()

def dfs(x, seq):
    visited.add(x)
    seq.append(x)
    
    if len(seq) == M:
        print(*seq)
        
    for v in range(x+1, N+1):
        if v not in visited:
            dfs(v, seq)

    visited.remove(x)
    seq.pop()
            
for i in range(1, N+1):
    dfs(i, [])
